src/components/search-bar/search-bar.jsx

```diff
@@ -16,14 +18,34 @@ import './search-bar.css';
 export default class SearchBar extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      query: {
+        url: null,
+        startDate: null,
+        endDate: null
+      }
+    };
     this._didSearch = this._didSearch.bind(this);
     this._dispatchSearch = debounce(this._dispatchSearch.bind(this), 500);
+    this.urlSearch = this.urlSearch.bind(this);
+    this.dateSearch = this.dateSearch.bind(this);
+  }
+
+  componentDidUpdate(previousProps, previousState) {
+    if (previousState.query !== this.state.query) {
+      this.props.onSearch(this.state.query);
+    }
```

Comment:
Depending how we manage state, this might not be an adequate check. (In the current setup, we should really be setting state with a function, and possibly merging objects there, which would mean we don’t always have a new object.)